test/precompile/native/evm-compatibility-ecrecover/EcrecoverCheck.js

```diff
@@ -0,0 +1,89 @@
+/*-
+ *
+ * Hedera JSON RPC Relay - Hardhat Example
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const { expect } = require('chai');
+const utils = require('./utils');
+const htsUtils = require('../../hedera-token-service/utils');
+const hre = require('hardhat');
+const { PrivateKey } = require('@hashgraph/sdk');
+
+const testEcrecover = async (address, initialKey, withAlias, changedTo) => {
+  const network = htsUtils.getCurrentNetwork();
+  const operatorId = hre.config.networks[network].sdkClient.operatorId;
+  const operatorKey = PrivateKey.fromStringECDSA(
+    hre.config.networks[network].sdkClient.operatorKey.replace('0x', '')
+  );
+  const client = await htsUtils.createSDKClient(operatorId, operatorKey);
+  const account = await utils.createAccount(client, initialKey, withAlias);
+  const initialResult = await utils.verifyEcrecover(
+    address,
+    client.setOperator(account.accountId, account.privateKey),
+    utils.formatPrivateKey(account.privateKey)
+  );
+  if (changedTo === '' || !initialResult) {
+    // Changing keys to the new ones was not requested, we can quit here.
+    return initialResult;
+  }
+  account.privateKey = await utils.changeAccountKeys(account, changedTo);
+
+  return await utils.verifyEcrecover(
+    address,
+    client.setOperator(account.accountId, account.privateKey),
+    utils.formatPrivateKey(account.privateKey)
+  );
+};
+
+describe('EcrecoverCheck', function () {
+  let address;
+  before(async () => {
+    address = await utils.deploy();
+  });
+
+  describe('Deployment', function () {
+    it('Should be deployed correctly', async function () {
+      expect(address).to.not.null;
+    });
+  });
+
+  describe('Verification', function () {
+    it('Ecrecover should work correctly for account with ECDSA key and EVM alias derived from ECDSA key.', async function () {
+      expect(await testEcrecover(address, 'ECDSA', true, '')).to.true;
```

Comment:
Could you convert these `to.true`/`to.false` to address comparison directly? This is to provide more contextual information when the test fails.

Essentially, perform the test of `utils.verifyEcrecover` directly in the `expect`.

Reply:
@acuarica I have implemented the requested changes. String comparison is now utilized in the test cases.